eZ/Publish/Core/FieldType/RichText/XSLTProcessorExtension.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace eZ\Publish\Core\FieldType\RichText;
+
+/**
+ * Helper functions available in XSL documents.
+ */
+class XSLTProcessorExtension
+{
+    const URL_PATTERN = '%^((ezlocation|ezurl|ezcontent):\/\/\d+(#[^\s]+)?)|(#[^\s]+)|(?:(?:https?|ftp)://)(?:\S+(?::\S*)?@|\d{1,3}(?:\.\d{1,3}){3}|(?:(?:[a-z\d\x{00a1}-\x{ffff}]+-?)*[a-z\d\x{00a1}-\x{ffff}]+)(?:\.(?:[a-z\d\x{00a1}-\x{ffff}]+-?)*[a-z\d\x{00a1}-\x{ffff}]+)*(?:\.[a-z\x{00a1}-\x{ffff}]{2,6}))(?::\d+)?(?:[^\s]*)?$%iu';
+
+    /**
+     * Return true if URL is valid.
+     *
+     * @param string $value
+     * @return bool
+     */
+    public static function isValidUrl($value)
```

Comment:
Note to reviewer: This method must be public if it is to be accessible from XSL

Reply:
yep, so `XSLTProcessorFuntions` would make that more clear :)

Reply:
Done.